src/main/scala/rules/Consumer.scala

```diff
@@ -320,16 +320,21 @@ object consumer extends ConsumptionRules {
 
         eval(s, eRcvr, pve, v)((s1, tRcvr, v1) =>
           eval(s1, ePerm, pve, v1)((s2, tPerm, v2) => {
-            val (relevantChunks, _) =
-              quantifiedChunkSupporter.splitHeap[QuantifiedFieldChunk](s2.h, BasicChunkIdentifier(field.name))
-            val (smDef1, smCache1) =
-              quantifiedChunkSupporter.summarisingSnapshotMap(
-                s2, field, Seq(`?r`), relevantChunks, v2)
-            v2.decider.assume(FieldTrigger(field.name, smDef1.sm, tRcvr))
-//            v2.decider.assume(PermAtMost(tPerm, FullPerm()))
+            val s2p = if (s2.heapDependentTriggers.contains(field)){
+              val (relevantChunks, _) =
+                quantifiedChunkSupporter.splitHeap[QuantifiedFieldChunk](s2.h, BasicChunkIdentifier(field.name))
+              val (smDef1, smCache1) =
+                quantifiedChunkSupporter.summarisingSnapshotMap(
+                  s2, field, Seq(`?r`), relevantChunks, v2)
+              v2.decider.assume(FieldTrigger(field.name, smDef1.sm, tRcvr))
```

Comment:
sorry for the noob question but I'm still getting into this part of the codebase. What does it mean to assume a trigger?

Reply:
When you write a field trigger { x.f } in Viper, on the SMT level, that'll be translated as triggering on a special function called something like loc_f. That's a boolean function whose value doesn't represent anything and which is just used in triggers. So here we assume that this function is true for some arguments, which adds the function application to the prover state, and will trigger any quantifiers that trigger on that field.